### ISSUE TITLE
Questions Regarding Case 1 Implementation in FlamePINN-1D

### ISSUE THREAD

**Issue by Mannava-Daasaradhi** [CAME-THU/FlamePINN-1D#2]
Hello,

I am working on implementing Case 1 (Freely Propagating Simple Flame) using FlamePINN-1D and want to ensure that my implementation exactly follows the methodology used in the research paper. I have a few questions regarding the setup and would really appreciate any guidance.

**Forward PINN (Solving for T, rho, u, YF, omega, p)**
Boundary Conditions:

Are Dirichlet conditions strictly enforced, or are they included as a soft loss term?

Should the temperature gradient (dT/dx) at x=0 be strictly enforced?

**Neural Network Architecture:**

The provided code uses a 3-layer MLP (64 neurons per layer, tanh activation). Would increasing the number of layers improve accuracy?

Would using sin activation instead of tanh improve performance for this case?

Physics-Informed Loss Function:

The energy equation includes omega (reaction rate). Should omega include a temperature threshold to prevent log(0) errors?

How is the loss weight for PDE constraints vs. data loss chosen?

Would it be beneficial to add pressure gradient constraints?

Sampling & Data Normalization:

The code uses Hammersley sampling for domain points. Is this the optimal approach, or would an alternative sampling strategy improve performance?

Should temperature T(x) be normalized to [-1,1] instead of using physical units?

Would extending Z-curve warm-up training beyond 1000 epochs improve convergence?

**Inverse PINN (Learning S_L from Data)**
How is S_L initialized in training?

Should we start with S_L = 0.4 m/s, or should it be randomly initialized?

Should we constrain S_L within a known range to prevent unphysical results?

Handling Noisy Data:

What noise level was used in the reference paper for T(x) and u(x)?

Should the PINN only learn from sparse points, or should it use the entire dataset?

PDE Constraints for S_L Learning:

How strongly should the PDE be enforced in the loss function?

Should we add a prior constraint based on empirical flame speed models?

Equivalence Ratio (phi) Scaling:

Should phi be normalized to [-1,1] for better network generalization?

I would really appreciate any guidance you could provide to ensure my implementation matches the expected results.

Thank you for your time.

Forward

`import torch
import deepxde as dde
import numpy as np
import matplotlib.pyplot as plt

# ✅ Device Configuration
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
print(f"✅ Using device: {device}")

# ✅ Constants & Parameters
R = 8.3145
A = 1.4e8
v = 1.6
Ea = 121417.2
W = 0.02897
lambda_ = 0.026
cp = 1000.0
qF = 5.0e7
T_in = 298.0
dTdx_in = 1e5
L = 1.5e-3
epsilon = 1e-6

# ✅ Neural Network Architecture (DeepXDE)
layer_sizes = [1, 64, 64, 64, 1]
activation = "tanh"
initializer = "Glorot normal"

# ✅ Define Domain & Geometry
geom = dde.geometry.Interval(0, L)

# ✅ Governing Equation (PDE for Temperature)
def flame_pde(x, T):
    dTdx = dde.grad.jacobian(T, x)
    d2Tdx2 = dde.grad.hessian(T, x)

    S_L = 0.4  # Fixed assumption for forward problem
    u = (S_L + R * T_in / (W * S_L) - torch.sqrt((S_L + R * T_in / (W * S_L))**2 - 4 * R * T / W)) / 2
    rho = S_L / (u + epsilon)
    YF = 0.4 / (4 + 0.4) + cp * (T_in - T) / qF
    omega = A * torch.exp(-Ea / (R * T)) * (rho * YF) ** v

    return rho * S_L * cp * dTdx - lambda_ * d2Tdx2 - omega * qF

# ✅ Boundary Conditions
def boundary_lhs(x, T):
    return T - T_in

def boundary_rhs(x, T):
    return dde.grad.jacobian(T, x) - dTdx_in

# ✅ Define PINN Model
data = dde.data.PDE(
    geom, flame_pde, 
    [dde.DirichletBC(geom, boundary_lhs, lambda x: np.isclose(x, 0)),
     dde.OperatorBC(geom, boundary_rhs, lambda x: np.isclose(x, 0))],
    num_domain=1001, num_boundary=2
)

net = dde.maps.FNN(layer_sizes, activation, initializer)
model = dde.Model(data, net)

# ✅ Warm-up Training
x_warmup = np.linspace(0, L, 100)[:, None]
T_warmup = np.linspace(T_in, 1800, 100)[:, None]
data_warm = dde.data.PDE(geom, lambda x, y: 0, [], num_domain=100)
model_warm = dde.Model(data_warm, net)

model_warm.compile("adam", lr=1e-3)
model_warm.train(epochs=1000)

# ✅ Full Training
model.compile("adam", lr=1e-3)
losshistory, train_state = model.train(epochs=30000)

# ✅ Plot Results
x_test = np.linspace(0, L, 1000)[:, None]
T_pred = model.predict(x_test)
plt.plot(x_test, T_pred, label="PINN Prediction")
plt.xlabel("x (m)")
plt.ylabel("Temperature (K)")
plt.legend()
plt.savefig("T_vs_x.png")
plt.show()
`

Inverse
`import deepxde as dde
import numpy as np
import matplotlib.pyplot as plt
import torch

# 🔥 Constants
R = 8.3145
A = 1.4e8
v = 1.6
Ea = 1.5e5
W = 0.02897
lam = 0.026
cp = 1000.0
qF = 5.0e7
T_in = 298.0
L = 1.5e-3
epsilon = 1e-6

# Define geometry
geom = dde.geometry.Interval(0, L)

# 🔄 Train inverse PINN for each φ
phi_values = np.array([0.4, 0.42, 0.44, 0.46, 0.48, 0.5])
sl_preds = []

for phi in phi_values:
    print(f"\n🚀 Training for phi = {phi:.2f}")

    # Trainable parameter: S_L
    S_L = dde.Variable(0.4)

    # Define PDE
    def pde(x, T):
        dTdx = dde.grad.jacobian(T, x)
        d2Tdx2 = dde.grad.hessian(T, x)

        c = S_L + R * T_in / (W * S_L)
        u = (c - dde.backend.sqrt(dde.backend.maximum(c**2 - 4 * R * T / W, epsilon))) / 2
        rho = S_L / (u + epsilon)
        YF = dde.backend.maximum(0.1 + cp * (T_in - T) / qF, 0.0)
        log_omega = np.log(A) - Ea / (R * T) + v * dde.backend.log(rho * YF + epsilon)
        omega = dde.backend.exp(dde.backend.minimum(log_omega, 20.0))

        return rho * S_L * cp * dTdx - lam * d2Tdx2 - omega * qF

    # Setup DeepXDE problem
    data = dde.data.PDE(
        geom, pde, [], num_domain=1001, num_boundary=2
    )

    # Neural Network
    net = dde.maps.FNN([1, 64, 64, 64, 1], "tanh", "Glorot normal")
    model = dde.Model(data, net)

    # Compile & train
    model.compile("adam", lr=1e-3, external_trainable_variables=[S_L])
    model.train(epochs=30000)

    # Store prediction
    sl_preds.append(S_L.value[0])
    print(f"✅ Phi = {phi:.2f}, Predicted S_L = {S_L.value[0]:.5f} m/s")

# 📊 Plot results
plt.plot(phi_values, sl_preds, "-o", label="PINN Predicted S_L")
plt.xlabel("Equivalence Ratio (Phi)")
plt.ylabel("Laminar Flame Speed (S_L) [m/s]")
plt.title("Inverse PINN: S_L vs Phi")
plt.grid(True)
plt.legend()
plt.savefig("inverse_deepxde_sl_vs_phi.png")
plt.show()
`

**Comment by JiahaoWu99** [comment]
1. Boundary Conditions: (1) Soft. (2) Yes. Otherwize, the problem will be ill-posed (because sL is unknown).
2. NN size and activation function can influence the performance. The conclusions must be made after testing.
3. Physics-Informed Loss Function: (1) Not required. Because T is constrained in [T_in, T_eq] by the output_transform of the NN. (2) 1:1 is OK for this case. As long as the ODEs and BCs are normalized. (3) What does the pressure gradient constraints mean?
4. Sampling & Data Normalization: (1) Adaptive sampling may improve the accuracy, but Hammersley is enough for this case. (2) T is constrained in [T_in, T_eq] to avoid unphysical solutions and divergence. (3) You can try. I guess not. Because the exact Z-curve is more far away from the true solution. 
5. Inverse PINN: (1) 0.4 m/s is appropriate. It is reasonable from experience. (2) You can try the randomly initialization. Our paper has given the results under different s_L_ini. (3) That is a good idea. You can try.
6. Handling Noisy Data: (1) See Fig.11. (2) Sparse. It is more practical and meaningful.
7. PDE Constraints for S_L Learning: (1) The ODE residuals should be low to obtain an accurate s_L. (2) That is a good idea. You can try.
8. Equivalence Ratio (phi) Scaling: Phi is not the input of NN. So, normalization is not needed. 
9. Your codes: (1) s_L should be unknown in forward probelms. (2) The DirichletBC may be used incorrectly. (3) Do you apply normalizations in NN input, NN output, ODEs and BCs?
